src/NuGet.Core/NuGet.Protocol/RemoteRepositories/HttpFileSystemBasedFindPackageByIdResource.cs

```diff
@@ -598,7 +601,31 @@ private class PackageInfo
 
             public string Path { get; set; }
 
-            public string ContentUri { get; set; }
+            public string BaseUri { private get; set; }
+            public string Id { private get; set; }
+
+            public string ContentUri
+            {
+                get
+                {
+                    var normalizedVersionString = Identity.Version.ToNormalizedString();
+                    string idInLowerCase = Id.ToLowerInvariant();
+
+                    var builder = StringBuilderPool.Shared.Rent(256);
```

Comment:
Roughly 50% of the string builders used in this codepath end up exceeding the 256 char limit, and thus aren't released back to the pool. I didn't address it in this PR, as delay calculating this ends up getting rid of the vast majority of the work, but it might be worth considering upping that 256 char limit used in StringBuilderPool.